File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> nums, int target)  {
        unordered_map<int, int> table;
        
        for (int i = 0; i < nums.size(); i++) {
            if (table.find(target - nums[i]) != table.end()) {
                return { i, table[target - nums[i]] };
            } else {
                table.insert({ nums[i], i });
            }
        }

        return {-1};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        start_index = 0
        end_index = 0

        for i in range(0, len(nums)):
            for j in range(0, len(nums)):
                if i != j and nums[i] + nums[j] == target:
                    start_index = i
                    end_index = j
                    break
        
        return [start_index, end_index]